source/Playnite/Database/GameDatabase_Filters.cs

```diff
@@ -840,5 +812,32 @@ private bool FilterByStyleOr(Game game, FilterSettings filterSettings)
 
             return true;
         }
+
+        private bool GetNameFilterResult(Game game, FilterSettings filterSettings)
+        {
+            if (filterSettings.Name.IsNullOrEmpty() || game.Name.IsNullOrEmpty())
+            {
+                return false;
+            }
+
+            if (filterSettings.NameSearchWithAcronyms &&
+                filterSettings.Name.IsStartOfStringAcronym(game.Name))
+            {
+                return true;
+            }
+            else if (filterSettings.Name.Length >= 2 && filterSettings.Name[0] == '^')

```

Comment:
This check should be done first because of explorer panel filter via name groups.